joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RubyNodeCreator.scala

```diff
@@ -656,10 +656,42 @@ class RubyNodeCreator extends RubyParserBaseVisitor[RubyNode] {
     ClassDeclaration(
       visit(ctx.classPath()),
       Option(ctx.commandOrPrimaryValue()).map(visit),
-      visit(ctx.bodyStatement())
+      lowerSingletonClassDeclarations(ctx.bodyStatement())
     )(ctx.toTextSpan)
   }
 
+  /** Lowers all MethodDeclaration found in SingletonClassDeclaration to SingletonMethodDeclaration.
+    * @param ctx
+    *   body context from class definitions
+    * @return
+    *   RubyNode with lowered MethodDeclarations where required
+    */
+  private def lowerSingletonClassDeclarations(ctx: RubyParser.BodyStatementContext): RubyNode = {
+    visit(ctx) match {
+      case stmtList: StatementList =>
+        StatementList(stmtList.statements.flatMap {
+          case singletonClassDeclaration: SingletonClassDeclaration =>
+            singletonClassDeclaration.baseClass match {
+              case Some(selfIdentifier: SelfIdentifier) =>
+                singletonClassDeclaration.body match {
+                  case singletonClassStmtList: StatementList =>
+                    singletonClassStmtList.statements.map {
+                      case method: MethodDeclaration =>
+                        SingletonMethodDeclaration(selfIdentifier, method.methodName, method.parameters, method.body)(
+                          method.span
+                        )
+                      case nonMethodStatement => nonMethodStatement
```

Comment:
Have you investigated if something like 
```ruby
 class << self
   x = 1
 end
```
Needs to be lowered to `self.x = 1`?

Reply:
If this is the case, we can leave a TODO and create a separate issue to handle it

Reply:
Created separate issue as instance and class members are not handled yet. #4246